cpp/src/map_test.cpp

```diff
@@ -114,6 +114,23 @@ void test_value() {
     ASSERT_THROWS(conversion_error, m.value(bad));
 }
 
+void test_copy_constructor() {
+    std::map<string, scalar> sm;
+    sm["a"] = 2;
+    sm["b"] = 3;
```

Comment:
Wouldn't this be clearer as:
`std::map<string, scalar> m = {{"a", 2}, {"b", 3}};`

Reply:
or even (syntax might not be 100% correct):
`auto m = std::map<string,scalar>{{"a", 2}, {"b", 3}};`